zipline/utils/events.py

```diff
@@ -236,6 +236,8 @@ def handle_data(self, context, data, dt):
 
 
 class EventRule(six.with_metaclass(ABCMeta)):
+    cal = None  # A calendar instance is set when scheduling a function.
```

Comment:
Maybe worth clarifying that the calendar is set on instances of the rule, as opposed to the class?

Reply:
Should we make `cal` an abstract property to ensure that all the subclasses implement setters?

Reply:
I thought about that as well. My only concern was about the performance of the properties inside the `should_trigger` implementation, but I didn't validate that, and `@lazyval` could also solve that.

Reply:
I think for simple attribute access `property `will actually be faster than `lazyval`.  lazyval has more complicated machinery in exchange for memoizing the result, but if the fetching the attribute is trivial, then I think property will probably have less overhead.